src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DkmUtilities.cs

```diff
@@ -45,17 +45,23 @@ private static IEnumerable<DkmClrModuleInstance> GetModulesInAppDomain(this DkmC
 
         internal unsafe static ImmutableArray<MetadataBlock> GetMetadataBlocks(this DkmClrRuntimeInstance runtime, DkmClrAppDomain appDomain)
         {
+            return runtime.GetMetadataBlocks(appDomain, ImmutableArray<MetadataBlock>.Empty);
+        }
+
+        internal unsafe static ImmutableArray<MetadataBlock> GetMetadataBlocks(this DkmClrRuntimeInstance runtime, DkmClrAppDomain appDomain, ImmutableArray<MetadataBlock> previousMetadataBlocks)
```

Comment:
You're right: this one is safe. GetMetadataBlock is unsafe. Thank you!

---
In reply to: [127064346](https://github.com/dotnet/roslyn/pull/20834#discussion_r127064346) [](ancestors = 127064346)